camel/embeddings/clip_embedding.py

```diff
@@ -0,0 +1,90 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from typing import Any, List, Union
+
+from PIL import Image
+
+from camel.embeddings import BaseEmbedding
+
+
+class CLIPEmbedding(BaseEmbedding):
```

Comment:
Better to use generics here to support typing system.
```suggestion
class CLIPEmbedding(BaseEmbedding[Union[str, Image.Image]]):
```

Reply:
Thanks for your review! I will change it.